cmd/beaker/main.go

```diff
@@ -153,3 +168,41 @@ func withSignal(parent context.Context) (context.Context, context.CancelFunc) {
 		cancel()
 	}
 }
+
+// login prompts the user for an authentication token, validates it,
+// and writes it to the configuration file.
+func login() error {
+	loginURL, err := url.Parse(beakerConfig.BeakerAddress)
+	if err != nil {
+		return err
+	}
+	loginURL.Path = path.Join(loginURL.Path, "user")
+
+	fmt.Println("You are not logged in. To log in, find your user token here:", loginURL.String())
+	fmt.Print("Enter your user token: ")
+	reader := bufio.NewReader(os.Stdin)
+	for {
+		input, err := reader.ReadString('\n')
+		if err != nil {
+			return err
+		}
+		beakerConfig.UserToken = strings.TrimSuffix(input, "\n")
```

Comment:
TrimSpace does a more thorough job of trimming leading and trailing whitespace.

```suggestion
		beakerConfig.UserToken = strings.TrimSpace(input)
```

Reply:
Fixed